### PR TITLE
fix(controller):unified paths to use project id

### LINKED ISSUE #1808: Unified project name validation logic with cilent
**Describe the bug**
the name of project are lower case(even is upper case, it would transfer it to lower) but the server keep the original name for project

**To Reproduce**

**Expected behavior**

**Environment**

### PR DESCRIPTION
**Pull request by goldenxinxing**
## Description

1. closes #1808 
2. fix wrong storage path 

## Modules
- [ ] UI
- [x] Controller
- [ ] Agent
- [x] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/api/_impl/data_store.py [hunk 1]
```diff
@@ -1408,6 +1407,40 @@ def get_data_store(instance_uri: str = "", token: str = "") -> DataStore:
...
+            "Content-Type": "application/json; charset=utf-8",
+            "Authorization": (
+                SWCliConfigMixed().get_sw_token(instance=instance_uri)
+                or os.getenv(SWEnv.instance_token, "")
+            ),
+        },
+        timeout=60,
+    )
+    resp.raise_for_status()
+    return resp.json().get("data", {}).get("id")
```
Comment (anda-ren) [inline client/starwhale/api/_impl/data_store.py:1441]: raise exception for `None`?

Comment (goldenxinxing) [reply client/starwhale/api/_impl/data_store.py]: done

**Review by anda-ren — DISMISSED** [review]
LGTM

File: client/starwhale/api/_impl/data_store.py [hunk 2]
```diff
@@ -1408,6 +1407,40 @@ def get_data_store(instance_uri: str = "", token: str = "") -> DataStore:
...
+        or type(project) == int
+    ):
+        return table_name_generator(project, table)
+    else:
+        return table_name_generator(
+            _get_remote_project_id(_instance_uri, project), table
+        )
+
+
+def _get_remote_project_id(instance_uri: str, project: Union[str, int]) -> Any:
```
Comment (tianweidut) [inline client/starwhale/api/_impl/data_store.py:1428]: add http_retry?

Comment (goldenxinxing) [reply client/starwhale/api/_impl/data_store.py:1429]: done

**Review by tianweidut — APPROVED** [review]
LGTM

**Pull request merged**